File: java/IEC62325/MarketOperations/CongestionRevenueRights/CRRSegment.java
package IEC62325.MarketOperations.CongestionRevenueRights;

import IEC61970.Base.Domain.Money;
import IEC61970.Base.Domain.DateTime;
import IEC61970.Base.Domain.Float;
import IEC61970.Base.Core.IdentifiedObject;

/**
 * CRRSegment represents a segment of a CRR in a particular time frame.
 * 
 * The segment class contains amount, clearing price, start date and time, end
 * date and time.
 * @created 03-Jan-2024 2:02:21 PM
 */
public class CRRSegment extends IdentifiedObject {

	/**
	 * Dollar amount = quantity x clearingPrice
	 */
	public Money amount;
	/**
	 * Clearing price of a CRR
	 */
	public Money clearingPrice;
	/**
	 * segment end date time
	 */
	public DateTime endDateTime;
	/**
	 * The MW amount associated with the CRR
	 */
	public Float quantity;
	/**
	 * segment start date time
	 */
	public DateTime startDateTime;

	public CRRSegment(){

	}
}//end CRRSegment

File: py/IEC62325/MarketOperations/CongestionRevenueRights/CRRSegment.py
from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from IEC61970.Base.Domain.DateTime import DateTime
from IEC61970.Base.Domain.Money import Money


class CRRSegment(IdentifiedObject):
    """
    CRRSegment represents a segment of a CRR in a particular time frame.

    The segment class contains amount, clearing price, start date and time, end
    date and time.
    """

    def __init__(self):
        super().__init__()  # Call to the superclass's constructor
        self.amount = Money()  # Dollar amount = quantity x clearingPrice
        self.clearing_price = Money()  # Clearing price of a CRR
        self.end_date_time = DateTime()  # segment end date time
        self.quantity = 1.0  # The MW amount associated with the CRR
        self.start_date_time = DateTime()  # segment start date time
